src/D2L.CodeStyle.Analyzers/ApiUsage/ServiceLocator/SingletonLocatorAnalyzer.cs

```diff
@@ -88,96 +71,96 @@ public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics
 			}
 		}
 
+		private delegate bool IsSingletonLocatorGet(
+			IMethodSymbol? methodSymbol
+		);
+
 		private delegate bool IsContainerType(
-				ITypeSymbol type,
-				[NotNullWhen( true )] out ITypeSymbol? containedType
-			);
+			ITypeSymbol type,
+			[NotNullWhen( true )] out ITypeSymbol? containedType
+		);
 
 		// Enforce that SingletonLocator can only load actual [Singleton]s
 		private static void EnforceSingletonsOnly(
-			SyntaxNodeAnalysisContext context,
-			Func<INamedTypeSymbol, bool> isSingletonLocator,
+			OperationAnalysisContext context,
+			IMethodSymbol methodSymbol,
+			IsSingletonLocatorGet isSingletonLocatorGet,
 			IsContainerType isContainerType
 		) {
-			var root = GetRootNode( context );
-			if( root == null ) {
+			if( !isSingletonLocatorGet( methodSymbol ) ) {
 				return;
 			}
-			var symbolinfo = context.SemanticModel.GetSymbolInfo( root, context.CancellationToken );
 
-			var method = symbolinfo.Symbol as IMethodSymbol;
+			ITypeSymbol typeArg = methodSymbol.TypeArguments.First();
+			if( isContainerType( typeArg, out ITypeSymbol? containedType ) ) {
+				typeArg = containedType;
+			}
 
-			if( method == null ) {
-				if( symbolinfo.CandidateSymbols == null ) {
-					return;
-				}
+			if( Attributes.Singleton.IsDefined( typeArg ) ) {
+				return;
+			}
 
-				if( symbolinfo.CandidateSymbols.Length != 1 ) {
-					return;
-				}
+			context.ReportDiagnostic( Diagnostic.Create(
+				descriptor: Diagnostics.SingletonLocatorMisuse,
+				location: context.Operation.Syntax.GetLocation(),
+				typeArg.GetFullTypeName()
+			) );
+		}
 
-				//This happens on method groups, such as
-				//  Func<IFoo> fooFunc = OldAndBrokenServiceLocator.Get<IFoo>;
-				method = symbolinfo.CandidateSymbols.First() as IMethodSymbol;
+		private static bool TryGetGetMethod(
+			Compilation compilation,
+			[NotNullWhen( true )] out IMethodSymbol? methodSymbol

```

Comment:
We've got nullable enabled, just use `null` rather than writing these things. The user can't forget to check `null` and then you won't need function names like `TryGetGetMethod`

Reply:
Yeah, I was gonna return `IMethodSymbol?`, but then I looked elsewhere and didn't. Happy to do